app/controllers/api/articles_controller.rb

```diff
@@ -32,11 +32,21 @@ def create
 
   def update
     article = Article.find(params[:id])
-    updated_article = article.update(article_params)
-    if updated_article
-      render json: { message: 'Your article has been successfully updated!' }, status: 200
+    if params[:published]
+      if current_user&.editor?
```

Comment:
There is a lot of nested if statements here. I thought you can take `if current_user&.editor?` outside `if params[:published]`, so to check it first If true - you just update the article in the else you can checkif params have `:published`. Also It feels like there is a way to do it with case statements, but I don't know

Reply:
Thought you would like that :(